### DISCUSSION TITLE
Remote Start

### DISCUSSION THREAD

**Discussion by overkillthemighty** [meshtastic/Discussions#118] [General]
For quite some time I have wanted a good reliable way to remote start my truck, especially in the winter. I started looking at meshtastic as an option a few months ago and have a few nodes up in my community now. I have an EVO ALL remote start system installed already, but it relies on the OEM key fob, so it is only useful if I am in line of sight and within a hundred feet. In my research I found that meshtastic uses encryption, but is susceptible to replay attacks, which was a little disappointing at first, as I was going to connect a GPIO to the wire on the unit that will trigger a start, but then I remembered coming across a project called EvoLink for allowing an Arduino to talk to the EVO unit and it all clicked!  

My plan is to use a 3.3v Arduino as a go-between to allow remote start as well as locking and unlocking. It should allow some lighting control as well as coolant temp readouts as well. I will be using a secure channel for communication that is set as default on the unit in the truck, and the serial module with text msg mode enabled. To address replay attacks, I am going to use a handshake model for the remote start and locks that goes like this: 

Send a command to "wake" the Arduino up.
It replies with a list of commands and a randomly generated number that is valid for 60 seconds.
I send the command I want along with the number
It confirms the command was received.  

In theory this should be relatively secure from what I understand of the system. I will post updates (I just got the Arduino doing the handshake over my channel, going to get it connected to the EVO and try sending the commands when I have time to get into the dash.)

**Comment by JSkier21** [comment]
How did this play out? 

I've got a Compustar aftermarket unit (hated the connected service through the manufacturer, hardly worked, and overpriced). 

Same issue with limited range on the fob, although the included ones work farther than the OEM. 

I'm surprised to not find much out there on remote starters and Meshtastic, it's a great idea.
